src/views/EditCheckinView/index.tsx

```diff
@@ -0,0 +1,137 @@
+'use client';
+import CheckInForm from '@/components/CheckInForm';
+import LoadingButton from '@/components/LoadingButton';
+import useSnackbar from '@/hooks/useSnackbar';
+import useValidation from '@/hooks/useValidation';
+import { EventResponse } from '@/types/dataModel/event';
+import { EventVolunteerResponse } from '@/types/dataModel/eventVolunteer';
+import { OrganizationResponse } from '@/types/dataModel/organization';
+import { CheckInFormData, zCheckInFormData } from '@/types/forms/checkIn';
+import { ValidationErrors } from '@/utils/validation';
+import { Typography } from '@mui/material';
+import Grid2 from '@mui/material/Unstable_Grid2';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+import useRoleConfirmation from '@/hooks/useRoleConfirmation';
+
+interface EditCheckInViewProps {
+  event: EventResponse;
+  organizations: OrganizationResponse[];
+  eventVolunteer: EventVolunteerResponse;
+}
+
+export default function EditCheckInView(props: EditCheckInViewProps) {
+  const [formData, setFormData] = useState<CheckInFormData>({
+    firstName: props.eventVolunteer.volunteer.firstName,
+    lastName: props.eventVolunteer.volunteer.lastName,
+    email: props.eventVolunteer.volunteer.email,
+    phoneNumber: props.eventVolunteer.volunteer.phoneNumber,
+    address: props.eventVolunteer.volunteer.address,
+    volunteerId: props.eventVolunteer.volunteer._id,
+    role: props.eventVolunteer.role,
+    organization: props.eventVolunteer.organization,
+  } as CheckInFormData);
+  const [validationErrors, setValidationErrors] = useState<
+    ValidationErrors<CheckInFormData> | undefined
+  >(undefined);
+  const [submitDisabled, setSubmitDisabled] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const validate = useValidation(zCheckInFormData);
+  const router = useRouter();
+  const confirmRole = useRoleConfirmation();
+
+  const { showSnackbar } = useSnackbar();
+  setLoading;
+  setValidationErrors;
+
+  // call the PUT API at src/app/api/events/check-in/[eventVolunteerId]
+  const onEditCheckIn = async () => {
+    setLoading(true);
+
+    const validationResult = validate(formData);
+    if (validationResult) {
+      setValidationErrors(validationResult);
+      setLoading(false);
+      return;
+    }
+
+    setValidationErrors(undefined);
+
+    let verifier: string | null = null;
+    if (
+      formData.role !== 'Food' &&
+      formData.role !== props.eventVolunteer.role
```

Comment:
not quite, it should be like 
```typescript
!props.eventVolunteer.volunteer.roleVerifications.find(
          (rv) => rv.role === formData.role
        )
```

Reply:
wait, but aren't we checking if the volunteer already has a role, and if they do we don't ask for verification again?